Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl

```diff
@@ -185,16 +186,66 @@ namespace AZ::IO::Internal
     {
         const size_t maxCharsToCompare = (AZStd::min)(left.size(), right.size());
 
-        int charCompareResult = exactCaseCompare
-            ? maxCharsToCompare ? strncmp(left.data(), right.data(), maxCharsToCompare) : 0
-            : maxCharsToCompare ? azstrnicmp(left.data(), right.data(), maxCharsToCompare) : 0;
-        return charCompareResult == 0
-            ? static_cast<int>(aznumeric_cast<ptrdiff_t>(left.size()) - aznumeric_cast<ptrdiff_t>(right.size()))
-            : charCompareResult;
+        if (exactCaseCompare)
+        {
+            const int charCompareResult = maxCharsToCompare
+                ? AZStd::char_traits<char>::compare(left.data(), right.data(), maxCharsToCompare)
+                : 0;
+            return charCompareResult == 0
+                ? static_cast<int>(static_cast<ptrdiff_t>(left.size()) - static_cast<ptrdiff_t>(right.size()))
+                : charCompareResult;
+        }
+        else
+        {
+            if (az_builtin_is_constant_evaluated())
+            {
+                // compile time implementation
+                auto ToLower = [](const char element) constexpr -> char
+                {
+                    return element >= 'A' && element <= 'Z' ? (element - 'A') + 'a' : element;
+                };
+                const auto mismatchResult = AZStd::ranges::mismatch(left, right, AZStd::char_traits<char>::eq, ToLower, ToLower);
+                const size_t leftMismatchCount = AZStd::ranges::distance(mismatchResult.in1, left.end());
+                const size_t rightMismatchCount = AZStd::ranges::distance(mismatchResult.in2, right.end());
+                if (leftMismatchCount == 0 && rightMismatchCount == 0)
+                {
+                    // Both path segments are equal return 0
+                    return 0;
+                }
+                else
+                {
+                    // There is a mismatch, so determine if it was due to characters not comparing equal
+                    if (leftMismatchCount != 0 && rightMismatchCount != 0)
+                    {
+                        return ToLower(*mismatchResult.in1) < ToLower(*mismatchResult.in2) ? -1 : 1;
+                    }
+                    else if (leftMismatchCount != 0)
+                    {
+                        // The left path segment has more characters left so it is greater than the right
+                        return 1;
+                    }
+                    else
+                    {
+                        // The left path segment has less characters left so it is less than the right
+                        return -1;
+                    }
+                }
+
+            }
+            else
+            {
+                const int charCompareResult = maxCharsToCompare
+                    ? azstrnicmp(left.data(), right.data(), maxCharsToCompare)
```

Comment:
_Super_ nitpick: missing the `// run time implementation` comment in this case. It's further from the branch than the other too, so might have to look up... just a little.

Reply:
Noted